### PR TITLE
Add Game Leaderboard + Some other features

### PR DESCRIPTION
**Pull request by TheShadowEevee**
Draft Notice: This PR is still a draft. As the branch this code came from is 6+ months behind the current master there are quite a few files and/or lines of code that were removed or changed that are still in that branch. Opening this as a draft makes it easier to remove this legacy code.

# About this PR
This PR adds a few features and fixes (Listed below) with the main addition of leaderboards for the most played games.

## New Features/Fixes (All done by @bennyman123abc)
 - Most played game leaderboards
 - Basic (Essentially empty) Admin Page
 - Admin role support (As an array in config.json, uses full Discord ID)
 - Fixes 404 pages showing "Login" even when the user is logged in
 - Potentially more?

### Contributions by @TheShadowEevee 
Little to none; I tested it, fixed a minor bug, and brought everything up-to-date with current master.

## Todo before pull
- Done

## Notes
- Game Leaderboard
  - Once the Game Leaderboards are released, they will only track from their initialization. They cannot track previous gameplay and due to the way it was tracked before I don't think it's possible.
  - Each region of a game will have it's own cover on the leaderboard.
  - Non-Wii games *should* work as well according to my testing.
- Admin Page
  - This uses one of the two possible ways I saw to add an admin page, the other being a "roles" array in each users .json which would allow for addition of other potential roles. I prefer the config.json method for just having an admin role, and the roles array method if more than one role could exist.
  - A refresh button exists on users tags but does not work.
- Other Notes
  - I got distracted while writing this so I may be forgetting something lmao

### CODE REVIEW

**Comment by TheShadowEevee** [comment]
This PR should be clear, though since it's decently sized I want to get the OK from someone else before merging.

TL;DR I'm about to DM larsen don't be surprised if one of us pulls this without another word on this PR lmao
Also please Squash commit this if someone else merges it, there's quite a few dumb commits

**Comment by larsenv** [comment]
Awesome!
